test/util/vectorization_test.py

```diff
@@ -315,5 +315,40 @@ def simple_func(x):
     assert simple_func(val_2) == 1
 
 
+def test_vectorize_callable_class():
+
+    # Test vectorization in 1d without data type --> lazy vectorization
+    arr = (np.arange(5) - 2)[None, :]
```

Comment:
Hard to see the slight difference between the two ranges. Better use `[-2, -1, 0, 1, 2]` explicitly.